flink-java/src/main/java/org/apache/flink/api/java/operators/SingleInputUdfOperator.java

```diff
@@ -54,8 +54,11 @@
 
 	private Map<String, DataSet<?>> broadcastVariables;
 
+	// NOTE: only set this variable via setSemanticProperties()
```

Comment:
Just a quick question (haven't checked the code). Does the analyzer also respect semantic information provided via the Operator API (withForwardedFields()), i.e., not via function annotations?